Sources/Target/Windows/Process.cpp

```diff
@@ -259,7 +259,21 @@ ErrorCode Process::wait() {
       continue;
     }
 
-    case EXCEPTION_DEBUG_EVENT:
+    case EXCEPTION_DEBUG_EVENT: {
+      // First-chance exceptions may yet be caught by the debugee, so don't break for them.
+      auto code = de.u.Exception.ExceptionRecord.ExceptionCode;
+      if (de.u.Exception.dwFirstChance &&
+          (code == DS2_EXCEPTION_UNCAUGHT_COM || code == DS2_EXCEPTION_UNCAUGHT_USER || code == DS2_EXCEPTION_UNCAUGHT_WINRT)) {
+        DS2LOG(Error, "Ignoring first-chance exception and continuing; code: %s",
+               Stringify::ExceptionCode(de.u.Exception.ExceptionRecord.ExceptionCode));
+        BOOL result = ::ContinueDebugEvent(pid(), de.dwThreadId, DBG_EXCEPTION_NOT_HANDLED);
+        if (!result) {
+          return Platform::TranslateError();
+        }
```

Comment:
I'd just collapse this:

```suggestion
        if (!::ContinueDebugEvent(pid(), de.dwThreadId, DBG_EXCEPTION_NOT_HANDLED))
          return Platform::TranslateError();
```

Reply:
Done. This is my preference as well, but I was following the existing art in these files which seems to pull out a temporary.
(I believe some people this makes code more debuggable but IMO that optimizes for the wrong thing)